src/Mutate.tsx

```diff
@@ -16,7 +16,10 @@ export interface States<TData, TError> {
   error?: GetState<TData, TError>["error"];
 }
 
-export type MutateMethod<TData, TRequestBody> = (data?: TRequestBody) => Promise<TData>;
+export type MutateMethod<TData, TRequestBody> = (
+  data: TRequestBody,
```

Comment:
`data` is not anymore optional, so we can define `string` for DELETE operation.
This is to have a bit more type safety, `mutate()` is not possible anymore if a body is required.